lib/etcdv3/kv.rb

```diff
@@ -28,11 +30,15 @@ def transaction(block)
         success: generate_request_ops(txn.success),
         failure: generate_request_ops(txn.failure)
       )
-      @stub.txn(request)
+      @stub.txn(request, **request_args)
     end
 
     private
 
+    def request_args
+      {metadata: @metadata}.merge(@timeout ? {deadline: Time.now.to_f + @timeout} : {})
```

Comment:
That's true. We were trying to preserve the original behavior of letting the GRPC lib deal with it, but if you don't think that's worthwhile, it would be nice to always have /something/ passed in.

Reply:
Right on.  My initial thoughts are that given the internal logic is hidden from the client, it may be beneficial to expose a default value on our end.  I figure it's an easy way to document the default behavior and make things more convenient on our end. :)